### ISSUE TITLE
Batch rename limitations are undocumented

### ISSUE THREAD

**Issue by jlneder** [jarun/nnn#392]
Hi, I tried to batch rename some files but nnn simply didn't take into account that i was reusing a filename and replaced one of the files with another. I was a little alarmed, and I decided to see if I had seen it wrong but it failed in the same way again.
When I saw the implementation i was alarmed:

`paste -d'\n' %s %s | sed 'N; /^\\(.*\\)\\n\\1$/!p;d' | tr '\n' '\\0' | xargs -0 -n2 mv 2>/dev/null`

On the plus side, It's simple and relatively clean. But It doesn't take into account for rename loops, swaps, replaces, etc.
The documentation should warn this, at least, but it would be better if it wasn't this way at all. 
An option would be to use "qmv" from renameutils. It is a complete batch rename program and is really lightweight.
It would be less code to simply use "qmv" but it will be one more dependency.
I see now that vidir was used before.  I like qmv a little more and it has the advantage that it only have readline as a dependency.

**Comment by 0xACE** [comment]
Hey there!

My understanding of `nnn`'s development plans is to rid itself of external dependencies as much as possible, in hope to make it portable, as in, bring it with you to any machine and make it work. Hence occurrences of such tendencies may appear.

Therefore it seems unlikely `nnn` will revert back to a external dependency such as `vidir` or `qmv`. Patches for fixing this may be appreciated though.

Although I would have to defer to one of the other devs who knows about this more than I do.

TL;DR: My post is nonsense.

**Comment by jarun** [comment]
> I tried to batch rename some files but nnn simply didn't take into account that i was reusing a filename and replaced one of the files with another.

Kindly provide the exact steps so we can reproduce this. @KlzXS can you please take a look once we have the info?

> The documentation should warn this

Thanks for the suggestion. We'll update the documentation and mention the alternatives. We missed this earlier.

> An option would be to use "qmv" from renameutils. It is a complete batch rename program and is really lightweight.

> I see now that vidir was used before.

Please add any of these as a plugin, set your custom keybind and you are ready to go.

**Comment by jlneder** [comment]
Say you have this files on a directory:
>file_1
>file_2

Press r, go to editor of choice and change the first to the second:

>file_2
>anything

When nnn run:

>paste -d'\n' %s %s | sed 'N; /^\\(.*\\)\\n\\1$/!p;d' | tr '\n' '\\0' | xargs -0 -n2 mv 2>/dev/null

It does not matter the name **_file_2_** is changed at all.
**_file_1_** would replace **_file_2_** on the first mv of xargs and then if  **_file_2_** name is changed xargs would proceed to do **_mv 'file_2' 'anything'_** .

I would see about making a plugin now. Thanks!

**Comment by jarun** [comment]
> It does not matter the name file_2 is changed at all.

Got it! I don't think we are going to fix it when there's are several alternative to add as plugins. There's also the vim plugin [vim-renamer](https://github.com/qpkorr/vim-renamer).

Will document the current implementation right away.

> I would see about making a plugin now

Head here:
